### PR TITLE
feat: made location as a default positional argument

### PR DESCRIPTION
**Pull request by capri-xiyue**
made location as a default positional argument

### CODE REVIEW

File: templates/commands/goldentest/new_test_cli.go [hunk 1]
```diff
@@ -48,11 +48,13 @@ func (c *NewTestCommand) Desc() string {
 
 func (c *NewTestCommand) Help() string {
 	return `
-Usage: {{ COMMAND }} [options] <test_name>
+Usage: {{ COMMAND }} [options] <test_name> <location>
```
Comment (drevell) [inline templates/commands/goldentest/new_test_cli.go:51]: I think the convention would be to add square brackets, like `[<location>]`, for things that are optional. We'd need to do this for `record` and `verify` too.

File: templates/commands/goldentest/new_test_flags.go [hunk 2]
```diff
@@ -112,4 +100,16 @@ func (r *NewTestFlags) Register(set *cli.FlagSet) {
 		}
 		return nil
 	})
+
+	// Default Location to the second CLI argument, if given
+	// If not given, default to current directory.
+	set.AfterParse(func(existingErr error) error {
+		r.Location = strings.TrimSpace(set.Arg(1))
+
+		if r.Location == "" {
```
Comment (drevell) [inline templates/commands/goldentest/new_test_flags.go:109]: Is there a way to test this? Some examples of flag testing are in abc_test.go and commands/render/render_test.go.

**Review by drevell — CHANGES_REQUESTED** [review]

File: templates/commands/goldentest/new_test_cli.go (hunk 1 above)
Comment (capri-xiyue) [reply templates/commands/goldentest/new_test_cli.go:51]: done

File: templates/commands/goldentest/new_test_flags.go (hunk 2 above)
Comment (capri-xiyue) [reply templates/commands/goldentest/new_test_flags.go:109]: done

**Review by drevell — APPROVED** [review]

File: templates/commands/goldentest/new_test_cli_test.go [hunk 3]
```diff
@@ -238,3 +239,74 @@ builtin_vars:
...
+			var cmd NewTestCommand
+			cmd.SetLookupEnv(cli.MapLookuper(nil))
+
+			err := cmd.Flags().Parse(tc.args)
+			if err != nil || tc.wantErr != "" {
+				if diff := testutil.DiffErrString(err, tc.wantErr); diff != "" {
+					t.Fatal(diff)
+				}
+				return
+			}
```
Comment (sethvargo) [inline templates/commands/goldentest/new_test_cli_test.go:306]: Do we want the nil check here? I think we might want to call `DiffErrString` without the surrounding conditional.

Comment (capri-xiyue) [reply templates/commands/goldentest/new_test_cli_test.go:306]: Thanks for catching it! I was blindly copy the existing test structure in abc goldentest. I sent another PR to fix all of them https://github.com/abcxyz/abc/pull/387

**Pull request merged**